__tests__/app.test.js

```diff
@@ -0,0 +1,225 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const testData = require("../db/data/test-data/index");
+const seed = require("../db/seeds/seed");
+const endpointsData = require("../endpoints.json");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+//Ticket 2
+describe("app", () => {
+  describe("get /api/topics", () => {
+    test("return request 200: returns status code", () => {
+      return request(app).get("/api/topics").expect(200);
+    });
+    test("return topics", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then((response) => {
+          const { topics } = response.body;
+          expect(Array.isArray(topics)).toBe(true);
+          expect(topics.length).toBe(3);
+          topics.forEach((topic) => {
+            expect(topic).toMatchObject({
+              slug: expect.any(String),
+              description: expect.any(String),
+            });
+          });
+        });
+    });
+  });
+});
+
+//Ticket 3
+describe("app", () => {
+  test("should return an object describing all available endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
+        expect(response.body).toEqual(endpointsData);
+      });
+  });
+});
+
+//Ticket 4
+describe("app", () => {
+  describe("GET /api/articles/:article_id", () => {
+    test("return request 200: returns status code", () => {
+      const articleID = 1;
+      return request(app).get(`/api/articles/${articleID}`).expect(200);
+    });
+
+    test("should return an article by its id", () => {
+      const articleID = 1;
+      const expectedArticle = {
+        article_id: 1,
+        title: "Living in the shadow of a great man",
+        topic: "mitch",
+        author: "butter_bridge",
+        body: "I find this existence challenging",
+        created_at: "2020-07-09T20:11:00.000Z",
+        votes: 100,
+        article_img_url:
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+      };
+      return request(app)
+        .get(`/api/articles/${articleID}`)
+        .expect(200)
+        .then((response) => {
+          const { article } = response.body;
+          expect(article).toMatchObject(expectedArticle);
+        });
+    });
+
+    test("returns 400 status for invalid input", () => {
+      const article_id = "Hello";
+      return request(app).get(`/api/articles/${article_id}`);
+      expect(res.status).toBe(400);
+      expect(res.body.message).toBe("Bad Request");
```

Comment:
I don't think this test is ever running these `expect` lines, because they are not inside of a `.then` which should follow on from the `.get(....)` , so this test is likely always passing no matter what. 
Refer to your test on _Line 76_ and have a look at how you're using `.then` on that test after the `.get`

Reply:
done